src/snowflake/snowpark/mock/_telemetry.py

```diff
@@ -0,0 +1,206 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+import atexit
+import json
+import logging
+import os
+import uuid
+from datetime import datetime
+from enum import Enum
+from typing import Optional
+
+from snowflake.connector.compat import OK
+from snowflake.connector.secret_detector import SecretDetector
+from snowflake.connector.telemetry_oob import REQUEST_TIMEOUT, TelemetryService
+from snowflake.connector.vendored import requests
+from snowflake.snowpark._internal.utils import (
+    get_os_name,
+    get_python_version,
+    get_version,
+)
+
+logger = logging.getLogger(__name__)
+
+OS_VERSION = get_os_name()
+PYTHON_VERSION = get_python_version()
+SNOWPARK_PYTHON_VERSION = get_version()
+
+TELEMETRY_VALUE_SNOWPARK_EVENT_TYPE = "Snowpark Python"
+TELEMETRY_KEY_TYPE = "Type"
+TELEMETRY_KEY_LOWER_TYPE = "type"
+TELEMETRY_KEY_UUID = "UUID"
+TELEMETRY_KEY_CREATED_ON = "Created_on"
+TELEMETRY_KEY_MESSAGE = "Message"
+TELEMETRY_KEY_TAGS = "Tags"
+TELEMETRY_KEY_PROPERTIES = "properties"
+TELEMETRY_KEY_SNOWPARK_VERSION = "Snowpark_Version"
+TELEMETRY_KEY_OS_VERSION = "OS_Version"
+TELEMETRY_KEY_PYTHON_VERSION = "Python_Version"
+TELEMETRY_KEY_EVENT_TYPE = "Event_type"
+TELEMETRY_KEY_CONN_UUID = "Connection_UUID"
+TELEMETRY_KEY_FEATURE_NAME = "feature_name"
+TELEMETRY_KEY_PARAMETERS_INFO = "parameters_info"
+TELEMETRY_KEY_ERROR_MESSAGE = "error_message"
+TELEMETRY_KEY_IS_INTERNAL = "is_internal"
+
+
+def generate_base_oob_telemetry_data_dict(
+    connection_uuid: Optional[str] = None,
+) -> dict:
+    return {
+        TELEMETRY_KEY_TYPE: TELEMETRY_VALUE_SNOWPARK_EVENT_TYPE,
+        TELEMETRY_KEY_UUID: str(uuid.uuid4()),
+        TELEMETRY_KEY_CREATED_ON: str(datetime.now().strftime("%Y-%m-%d %H:%M:%S")),
+        TELEMETRY_KEY_MESSAGE: {TELEMETRY_KEY_CONN_UUID: connection_uuid},
+        TELEMETRY_KEY_TAGS: {
+            TELEMETRY_KEY_SNOWPARK_VERSION: SNOWPARK_PYTHON_VERSION,
+            TELEMETRY_KEY_OS_VERSION: OS_VERSION,
+            TELEMETRY_KEY_PYTHON_VERSION: PYTHON_VERSION,
+        },
+        TELEMETRY_KEY_PROPERTIES: {
+            TELEMETRY_KEY_LOWER_TYPE: TELEMETRY_VALUE_SNOWPARK_EVENT_TYPE
+        },
+    }
+
+
+class LocalTestTelemetryEventType(Enum):
+    UNSUPPORTED = "unsupported"
+    SUPPORTED = "supported"
+    SESSION_CONNECTION = "session"
+
+
+class LocalTestOOBTelemetryService(TelemetryService):
+    PROD = "https://client-telemetry.c1.us-west-2.aws.app.snowflake.com/enqueue"
```

Comment:
Does the client have to auth to access this endpoint?

Reply:
no, this endpoint is publicly available.
this is out-of-band service so anyone can access this endpoint, but the telemetry data has to comply with the format otherwise the service will ignore the request